dotCMS/src/main/java/com/dotcms/ai/client/openai/OpenAIClient.java

```diff
@@ -0,0 +1,166 @@
+        }
+
+        final JSONObject payload = jsonRequest.getPayload();
+        final String modelName = Optional
+                .ofNullable(payload.optString(AiKeys.MODEL))
+                .orElseThrow(() -> new DotAIModelNotFoundException("Model is not present in the request"));
+        // TODO: pr-split -> uncomment this line
+        //final Tuple2<AIModel, Model> modelTuple = appConfig.resolveModelOrThrow(modelName, jsonRequest.getType());
+        final Tuple2<AIModel, Model> modelTuple = Tuple.of(null, null);
+        final AIModel aiModel = modelTuple._1;
+
+        if (!modelTuple._2.isOperational()) {
+            AppConfig.debugLogger(
+                    getClass(),
+                    () -> String.format("Resolved model [%s] is not operational, avoiding its usage", modelName));
+            throw new DotAIModelNotOperationalException(String.format("Model [%s] is not operational", modelName));
+        }
+
+        final long sleep = lastRestCall.computeIfAbsent(aiModel, m -> 0L)
+                + aiModel.minIntervalBetweenCalls()
+                - System.currentTimeMillis();
+        if (sleep > 0) {
+            Logger.info(
+                    this,
+                    "Rate limit:"
+                            + aiModel.getApiPerMinute()
+                            + "/minute, or 1 every "
+                            + aiModel.minIntervalBetweenCalls()
+                            + "ms. Sleeping:"
+                            + sleep);
+            Try.run(() -> Thread.sleep(sleep));
+        }
+
+        lastRestCall.put(aiModel, System.currentTimeMillis());
+
+        try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+            final StringEntity jsonEntity = new StringEntity(payload.toString(), ContentType.APPLICATION_JSON);
+            final HttpUriRequest httpRequest = AIClient.resolveMethod(jsonRequest.getMethod(), jsonRequest.getUrl());
+            httpRequest.setHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON);
+            httpRequest.setHeader(HttpHeaders.AUTHORIZATION, "Bearer " + appConfig.getApiKey());
+
+            if (!payload.getAsMap().isEmpty()) {
+                Try.run(() -> ((HttpEntityEnclosingRequestBase) httpRequest).setEntity(jsonEntity));
+            }
+
+            try (CloseableHttpResponse response = httpClient.execute(httpRequest)) {
+                final BufferedInputStream in = new BufferedInputStream(response.getEntity().getContent());
+                final byte[] buffer = new byte[1024];
+                int len;
+                while ((len = in.read(buffer)) != -1) {
+                    output.write(buffer, 0, len);
+                    output.flush();
+                }
+            }
+        } catch (Exception e) {
+            if (appConfig.getConfigBoolean(AppKeys.DEBUG_LOGGING)){
+                Logger.warn(this, "INVALID REQUEST: " + e.getMessage(), e);
```

Comment:
should be using a diff logger type?

Reply:
This is the logging coming from `OpenAIRequest`. Going to use `AppConfig.loggerBug()` instead.

Reply:
Not really.